analyzer/src/main/kotlin/PackageManager.kt

```diff
@@ -90,22 +92,33 @@ abstract class PackageManager(
             FileSystems.getDefault().getPathMatcher("glob:**/$it")
         }
 
+        /**
+         * Constant for an [Excludes] object that does not define any excludes. This is used unless the analyzer is
+         * configured to skip excludes.
+         */
+        private val EMPTY_EXCLUDES = Excludes()
+
         /**
          * Recursively search the [directory] for files managed by any of the [packageManagers]. The search is performed
-         * depth-first so that root project files are found before any subproject files for a specific manager.
+         * depth-first so that root project files are found before any subproject files for a specific manager. Path
+         * excludes defined by the given [excludes] are taken into account; the corresponding directories are skipped.
          */
         fun findManagedFiles(
             directory: File,
-            packageManagers: Collection<PackageManagerFactory> = ALL.values
+            packageManagers: Collection<PackageManagerFactory> = ALL.values,
+            excludes: Excludes = EMPTY_EXCLUDES
```

Comment:
The `PackageManager` has an instance of `RepositoryConfiguration` containing all excludes.
So, in case `--skip-excluded` is passed, the excludes are passed redundantly. Would it make sense to pass a `Boolean` / flag instead?

Reply:
`findManagedFiles()` is a function of the companion object. So the `RepositoryConfiguration` is not available here.